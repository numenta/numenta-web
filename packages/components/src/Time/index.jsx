// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


/**
 *
 */
const Time = ({children, moment}) => (
  <time className={styles.time} dateTime={moment.format()}>
    {children}
  </time>
)

Time.propTypes = {
  children: React.PropTypes.node.isRequired,
  moment: React.PropTypes.object,
}

export default Time
