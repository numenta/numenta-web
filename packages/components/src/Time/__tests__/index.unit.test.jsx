import moment from 'moment'
import React from 'react'
import renderer from 'react-test-renderer'

import Time from '../../Time'

const when = '2016-09-21T18:10:32'


describe('Time React component', () => {

  it('Renders correctly', () => {
    const occur = moment(new Date(when)).utcOffset(0, false)
    const component = renderer.create(
      <Time moment={occur}>
        {when}
      </Time>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
