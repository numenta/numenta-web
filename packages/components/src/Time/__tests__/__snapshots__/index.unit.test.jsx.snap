exports[`Time React component Renders correctly 1`] = `
<time
  className="time"
  dateTime="2016-09-21T18:10:32Z">
  2016-09-21T18:10:32
</time>
`;
